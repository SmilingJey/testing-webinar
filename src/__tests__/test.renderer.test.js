import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../components/Card';
import store from "../store/store"

describe('Check Card component', () => {
  let mockCard = {
    "likes": [],
    "_id": "1",
    "name": "Deer",
    "link": "https://images.unsplash.com/",
    "owner":
    {
        "name": "John",
        "about": "Web-developer",
        "avatar": "https://images.unsplash.com/",
        "_id": "1",
        "cohort": "cohort-13"
    },
    "createdAt": "2021-10-24T19:41:51.239Z"
  };

  test('should be equal to snapshot', () => {
    const renderedValue = TestRenderer.create(<Provider store={store}>
      <Card card={mockCard} onImageClick={()=>{}} onDelete={()=>{}}/></Provider>);
    expect(renderedValue).toMatchSnapshot();
  })

  test('should be equal to shallow snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Provider store={store}>
      <Card card={mockCard} onImageClick={()=>{}} onDelete={()=>{}}/></Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  })
})