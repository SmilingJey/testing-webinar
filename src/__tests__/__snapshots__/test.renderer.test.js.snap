// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Card component should be equal to shallow snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "getServerState": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Card
    card={
      Object {
        "_id": "1",
        "createdAt": "2021-10-24T19:41:51.239Z",
        "likes": Array [],
        "link": "https://images.unsplash.com/",
        "name": "Deer",
        "owner": Object {
          "_id": "1",
          "about": "Web-developer",
          "avatar": "https://images.unsplash.com/",
          "cohort": "cohort-13",
          "name": "John",
        },
      }
    }
    onDelete={[Function]}
    onImageClick={[Function]}
  />
</Context.Provider>
`;

exports[`Check Card component should be equal to snapshot 1`] = `
<li
  className="places__item card"
>
  <div
    className="card__image"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(https://images.unsplash.com/)",
      }
    }
  />
  <button
    className="card__delete-button card__delete-button_hidden"
    onClick={[Function]}
    type="button"
  />
  <div
    className="card__description"
  >
    <h2
      className="card__title"
    >
      Deer
    </h2>
    <div
      className="card__likes"
    >
      <button
        className="card__like-button false"
        onClick={[Function]}
        type="button"
      />
      <p
        className="card__like-count"
      >
        0
      </p>
    </div>
  </div>
</li>
`;
